source/fe/mapping_q.cc

```diff
@@ -235,22 +239,25 @@ MappingQ<dim,spacedim>::get_face_data (const UpdateFlags update_flags,
   // fill the data of both the Q_p and the Q_1 objects in parallel
   Threads::TaskGroup<> tasks;
   tasks += Threads::new_task (std_cxx11::function<void()>
-                              (std_cxx11::bind(&MappingQ<dim,spacedim>::compute_face_data,
-                                               this,
+                              (std_cxx11::bind(&MappingQ1<dim,spacedim>::InternalData::initialize_face,
+                                               data,
                                                update_flags,
                                                std_cxx11::cref(q),
-                                               quadrature.size(),
-                                               std_cxx11::ref(*data))));
+                                               quadrature.size())));
   if (!use_mapping_q_on_all_cells)
     tasks += Threads::new_task (std_cxx11::function<void()>
-                                (std_cxx11::bind(&MappingQ<dim,spacedim>::compute_face_data,
-                                                 this,
+                                (std_cxx11::bind(&MappingQ1<dim,spacedim>::InternalData::initialize_face,
+                                                 &data->mapping_q1_data,
                                                  update_flags,
                                                  std_cxx11::cref(q),
-                                                 quadrature.size(),
-                                                 std_cxx11::ref(data->mapping_q1_data))));
+                                                 quadrature.size())));
   tasks.join_all ();
 
+  // TODO: parallelize this as well
+  compute_shapes (q.get_points(), *data);
+  if (!use_mapping_q_on_all_cells)
+    compute_shapes (q.get_points(), data->mapping_q1_data);
```

Comment:
Will do this in one of the next patches, by the way. I'd like to re-join the effect of `compute_shapes` into the (what is now) `initialize()` function.